### PR TITLE
Add flush_job_test to the list of ASSERT_STATUS_CHECKED tests

### PR DESCRIPTION
**Pull request by zhichao-cao**
test plan: pass ASSERT_STATUS_CHECKED=1 make -j48 flush_job_test

### CODE REVIEW

File: db/flush_job.cc [hunk 1]
```diff
@@ -130,6 +130,7 @@ FlushJob::FlushJob(
 }
 
 FlushJob::~FlushJob() {
+  io_status_.PermitUncheckedError();
```
Comment (akankshamahajan15) [inline db/flush_job.cc:133]: Can we add assert ```io_status()```  is ok in flush_job_test.cc instead of ```PermitUncheckedError()```?

Comment (mrambacher) [reply db/flush_job.cc:133]: I actually think how it is coded is correct and it should not be an assert.  If one simply creates a FlushJob and does nothing with it, then the assert would be triggered.

**Review by mrambacher — APPROVED** [review]
LGTM.

File: db/flush_job.cc (hunk 1 above)
Comment (zhichao-cao) [reply db/flush_job.cc:133]: @akankshamahajan15 I agree with what Mark said.

**Pull request closed without merging**